src/Columns/ColumnVector.cpp

```diff
@@ -496,6 +575,14 @@ ColumnPtr ColumnVector<T>::filter(const IColumn::Filter & filt, ssize_t result_s
     static constexpr size_t SIMD_BYTES = 64;
     const UInt8 * filt_end_aligned = filt_pos + size / SIMD_BYTES * SIMD_BYTES;
 
+#if USE_MULTITARGET_CODE
```

Comment:
Maybe pull out l. 586 - 622 into it's own function (similar to doFilterAligned()) but in "Default" namespace, i.e. sth. like:

```
DECLARE_DEFAULT_CODE(
inline void doFilterAligned(const UInt8 *& filt_pos, const UInt8 *& filt_end_aligned, const T *& data_pos, Container & res_data)
)

#if USE_MULTITARGET_CODE
    if (sizeof(T) == 1 || sizeof(T) == 2 || sizeof(T) == 4 || sizeof(T) == 8)
    {
    if (isArchSupported(TargetArch::AVX512VBMI2))
            TargetSpecific::AVX512VBMI2::doFilterAligned<T, Container, SIMD_BYTES>(filt_pos, filt_end_aligned, data_pos, res_data);
    }
    TargetSpecific::Default::doFilterAligned(...)
```

(I am referring to the technique mentioned in the documentation at the beginning of "Common/TargetSpecific.h".)